scripts/pre_commit_linter.py

```diff
@@ -3124,26 +3349,55 @@ def main(args=None):
 
     read_files(all_filepaths)
     categorize_files(all_filepaths)
-    linting_processes, result_queues, result_stdout = _lint_all_files(
-        _FILES['.js'], _FILES['.ts'], _FILES['.py'], _FILES['.html'],
-        _FILES['.css'], verbose_mode_enabled)
-    code_owner_message = _check_codeowner_file(verbose_mode_enabled)
-    # Pylint requires to provide paramter "this_bases" and "d", guess due to
-    # meta class.
-    js_ts_lint_checks_manager = JsTsLintChecksManager( # pylint: disable=no-value-for-parameter
-        verbose_mode_enabled)
-    other_lint_checks_manager = OtherLintChecksManager(   # pylint: disable=no-value-for-parameter
-        verbose_mode_enabled)
-    all_messages = code_owner_message
-    js_message = js_ts_lint_checks_manager.perform_all_lint_checks()
-    other_messages = other_lint_checks_manager.perform_all_lint_checks()
-    all_messages += js_message + other_messages
 
-    all_messages += _join_linting_process(
-        linting_processes, result_queues, result_stdout)
+    max_concurrent_runs = 1
+
+    # Prepare tasks.
+    concurrent_count = min(multiprocessing.cpu_count(), max_concurrent_runs)
+    semaphore = threading.Semaphore(concurrent_count)
+
+    tasks_custom = []
+    tasks_third_party = []
+    for file_extension_type in file_extension_types:
+        custom_linter = LintingTaskSpec(
+            _FILES['.js'], _FILES['.ts'], _FILES['.py'], _FILES['.html'],
+            _FILES['.css'], verbose_mode_enabled, file_extension_type)
+
+        third_party_linter = ThirdPartyLinter(
+            _FILES['.js'], _FILES['.ts'], _FILES['.py'], _FILES['.html'],
+            _FILES['.css'], verbose_mode_enabled, file_extension_type)
+
+        task_custom = semaphore_utils.create_task(
+            custom_linter.lint_all_files, verbose_mode_enabled,
+            name=file_extension_type)
+        task_third_party = semaphore_utils.create_task(
+            third_party_linter.lint_all_files, verbose_mode_enabled,
+            semaphore=semaphore, name=file_extension_type)
+        tasks_custom.append(task_custom)
+        tasks_third_party.append(task_third_party)
+
+    p1 = multiprocessing.Process(
+        target=semaphore_utils.execute_tasks,
+        args=(tasks_custom, ))
+    p2 = multiprocessing.Process(
+        target=semaphore_utils.execute_tasks,
+        args=(tasks_third_party, semaphore))
+
+    p1.start()
+    p2.start()
+
+    p1.join()
+    p2.join()
```

Comment:
This wasn't what I meant. You only want to kick off the other queue once the other queue is complete. Also, by doing this, you're going to spawn a few more child processes than is necessary. Instead, you want to call execute_tasks for one queue and then the other once that is done.

Reply:
@kevinlee12 so you want this
`sempahore_utils.execute_tasks(tasks_custom)
semaphore_utils.execute_tasks(tasks_third_party, semaphore=semaphore)` ?

Reply:
correct